nncf/torch/graph/graph_matching.py

```diff
@@ -44,20 +44,40 @@ def __and__(self, other):
     def _iterate_alternatives(self, nodes):
         return powerset(nodes, min_r=1)
 
-    def match(self, nodes, graph):
+    def _find_all_matches(self, nodes, graph):
         all_matches = []
         for n in self._iterate_alternatives(nodes):
             result = self._match(n, graph)
             if not result:
                 continue
+            if isinstance(result, list):
+                # https://github.com/PyCQA/pylint/issues/3105
+                # pylint: disable=E1133
+                for res in result:
+                    n, following = res
+                    following = list(following)
+                    if not isinstance(n, list):
+                        n = [n]
+
+                    all_matches.append((n, following))
+                # pylint: enable=E1133
+            else:
+                n, following = result
+                following = list(following)
+                if not isinstance(n, list):
+                    n = [n]
+
+                all_matches.append((n, following))
+        if not all_matches:
+            return None
```

Comment:
@aleksei-kashapov  Why we can not return an empty list?

Reply:
indeed, we can

Reply:
Let's do it. Please update your PR.

Reply:
Done